cmd/terramate/cli/run_script.go

```diff
@@ -0,0 +1,168 @@
+			color.BlueString(result.ScriptCfg.Range.String()),
+			color.BlueString(fmt.Sprintf("%d", len(result.ScriptCfg.Jobs))),
+		)
+
+		if len(result.Stacks) == 0 {
+			c.output.MsgStdOut(color.RedString("(skipping) "),
+				"provided criteria does not allow script execution",
+			)
+		}
+
+		for _, st := range result.Stacks {
+			ectx, err := scriptEvalContext(c.cfg(), st.Stack)
+			if err != nil {
+				logger.Fatal().Msgf("failed to get context: %s", err)
+			}
+
+			evalScript, err := config.EvalScript(ectx, *result.ScriptCfg)
+			if err != nil {
+				logger.Fatal().Msgf("failed to eval script: %s", err)
+			}
+
+			for jobNum, j := range evalScript.Jobs {
+				c.output.MsgStdOut("")
+				for cmdNum, cmd := range j.Commands() {
+					printCommand(c.stderr, cmd, st.Dir().String(), jobNum, cmdNum)
+
+					env, err := run.LoadEnv(c.cfg(), st.Stack)
+					if err != nil {
+						logger.Fatal().Msgf("failed to load env: %s", err)
+					}
+
+					if err := c.executeCommand(
+						cmd, st.Dir().HostPath(c.rootdir()), env, c.parsedArgs.Experimental.RunScript.DryRun,
+					); err != nil {
+						logger.Fatal().Msgf("unable to execute command: %s", err)
+					}
+					c.output.MsgStdOut("")
+				}
+			}
+		}
+	}
+}
+
+func (c *cli) executeCommand(cmd []string, wd string, env []string, dryRun bool) error {
+	if dryRun {
+		c.output.MsgStdOut(color.BlueString("# -- this is a dry run -- #"))
+		return nil
+	}
+
+	newCmd, err := makeCommand(cmd, wd, env, c.stdout, c.stderr)
+	if err != nil {
+		return fmt.Errorf("failed to prepare command: %s", err)
+	}
+
+	if err := newCmd.Run(); err != nil {
+		return fmt.Errorf("failed to execute command: %s", err)
+	}
+
+	return nil
+}
+
+func makeCommand(command []string, dir string, env []string, stdout, stderr io.Writer) (*exec.Cmd, error) {
+	cmdPath, err := exec.LookPath(command[0])
+	if err != nil {
+		return nil, fmt.Errorf("%s: command not found: %s", command[0], err)
+	}
+
+	runCmd := exec.Command(cmdPath, command[1:]...)
+	runCmd.Dir = dir
+	runCmd.Env = env
+	runCmd.Stdout = stdout
+	runCmd.Stderr = stderr
+
+	return runCmd, nil
+}
+
+// printCommand pretty prints the cmd and attaches a "prompt" style prefix to it
+// for example:
+// /somestack (job:0.0)> echo hello
+func printCommand(w io.Writer, cmd []string, wd string, jobNum, cmdNum int) {
+	prompt := color.GreenString(fmt.Sprintf("%s (job:%d.%d)>", wd, jobNum, cmdNum))
+	fmt.Fprintln(w, prompt, color.YellowString(strings.Join(cmd, " ")))
+}
+
+func scriptEvalContext(root *config.Root, st *config.Stack) (*eval.Context, error) {
+	globalsReport := globals.ForStack(root, st)
+	if err := globalsReport.AsError(); err != nil {
+		return nil, err
+	}
+
+	evalctx := eval.NewContext(stdlib.Functions(st.HostDir(root)))
+	runtime := root.Runtime()
+	runtime.Merge(st.RuntimeValues(root))
+	evalctx.SetNamespace("terramate", runtime)
+	evalctx.SetNamespace("global", globalsReport.Globals.AsValueMap())
+	evalctx.SetEnv(os.Environ())
```

Comment:
is this needed?

Reply:
This allows users to have the following command syntax which seems uesful to me:

```hcl
command = ["echo", "${env.SOME_VAR}"]
```

What do you think?

Reply:
yes let's add it .... if we discover it being bad, we can remove it as it is an experiment where we can break at any time ;)